paypal/express/views.py

```diff
@@ -388,6 +388,46 @@ def get_shipping_method(self, basket, shipping_address=None, **kwargs):
 
 class ShippingOptionsView(View):
 
+    def get(self, request, *args, **kwargs):
+        """
+        We use the shipping address given to use by PayPal to
+        determine the available shipping method
+        """
+        # Basket ID is passed within the URL path.  We need to do this as some
+        # shipping options depend on the user and basket contents.  PayPal do
+        # pass back details of the basket contents but it would be royal pain to
+        # reconstitute the basket based on those - easier to just to piggy-back
+        # the basket ID in the callback URL.
+        basket = get_object_or_404(Basket, id=kwargs['basket_id'])
+        user = basket.owner
+        if not user:
+            user = AnonymousUser()
+
+        logger.info(kwargs['country_code'])
+
+        # Create a shipping address instance using the data passed back
+        country_code = self.request.POST.get(
+            'SHIPTOCOUNTRY', None)
+        try:
+            country = Country.objects.get(iso_3166_1_a2=kwargs['country_code'])
+        except Country.DoesNotExist:
+            country = Country()
```

Comment:
Why wouldn't we do `Country.get_or_create` instead of this?

Reply:
Good call.

Reply:
Just tried this: when I use Country.get_or_create() an error is thrown later on 

```
Traceback (most recent call last):
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/srv/sites/gredacted/paypal/express/views.py", line 450, in post
    country=country
  File "/srv/sites/redacted/lib/python2.7/site-packages/django/db/models/base.py", line 555, in __init__
    _setattr(self, field.name, rel_obj)
  File "/srv/sites/gasworks.org.uk/lib/python2.7/site-packages/django/db/models/fields/related_descriptors.py", line 216, in __set__
    self.field.remote_field.model._meta.object_name,
ValueError: Cannot assign "(<Country: United Kingdom>, False)": "ShippingAddress.country" must be a "Country" instance.
```

Looking at the master branch for this this code isn't actually changed in my fork, it's exactly as in master, so perhaps you need to speak to the upstream maintainer about this logic?